File: app/src/main/java/cn/intret/app/picgo/model/event/RescanFolderThumbnailListMessage.java
package cn.intret.app.picgo.model.event;

import java.io.File;
import java.util.List;

/**
 * 目录的缩略图被更新时发送
 */
public class RescanFolderThumbnailListMessage {
    File mDirectory;
    List<File> mThumbnails;

    public File getDirectory() {
        return mDirectory;
    }

    public RescanFolderThumbnailListMessage setDirectory(File directory) {
        this.mDirectory = directory;
        return this;
    }

    public List<File> getThumbnails() {
        return mThumbnails;
    }

    public RescanFolderThumbnailListMessage setThumbnails(List<File> thumbnails) {
        mThumbnails = thumbnails;
        return this;
    }

    @Override
    public String toString() {
        return "RescanFolderThumbnailListMessage{" +
                "mDirectory=" + mDirectory +
                '}';
    }
}

File: app/src/main/java/cn/intret/app/picgo/model/event/RescanFolderThumbnailListMessage.kt
package cn.intret.app.picgo.model.event

import java.io.File

/**
 * 目录的缩略图被更新时发送
 */
class RescanFolderThumbnailListMessage {
    internal lateinit var mDirectory: File
    internal lateinit var mThumbnails: List<File>

    fun getDirectory(): File {
        return mDirectory
    }

    fun setDirectory(directory: File): RescanFolderThumbnailListMessage {
        this.mDirectory = directory
        return this
    }

    fun getThumbnails(): List<File> {
        return mThumbnails
    }

    fun setThumbnails(thumbnails: List<File>): RescanFolderThumbnailListMessage {
        mThumbnails = thumbnails
        return this
    }

    override fun toString(): String {
        return "RescanFolderThumbnailListMessage{" +
                "mDirectory=" + mDirectory +
                '}'.toString()
    }
}
